### ISSUE TITLE
 Add support for `GET` in `SET` command

### ISSUE THREAD

**Issue by arpitbbhayani** [DiceDB/dice#207]
We already have the `SET` command implemented. Now we need to add support for `GET` option similar to the [`GET` option in Redis](https://redis.io/docs/latest/commands/set/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

**Comment by yaten2302** [comment]
@arpitbbhayani , could you please guide that do we need more flags for the `GET` command, like here in redis: <br>
![image](https://github.com/user-attachments/assets/4b5ee02c-4834-4740-9153-2c08c8b501f2)

The `EX`, `PX` etc... Am I right?

**Comment by JyotinderSingh** [comment]
> @arpitbbhayani , could you please guide that do we need more flags for the `GET` command, like here in redis: ![image](https://private-user-images.githubusercontent.com/129659514/354658050-4b5ee02c-4834-4740-9153-2c08c8b501f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2NTU4MDEsIm5iZiI6MTcyMjY1NTUwMSwicGF0aCI6Ii8xMjk2NTk1MTQvMzU0NjU4MDUwLTRiNWVlMDJjLTQ4MzQtNDc0MC05MTUzLTJjMDhjOGI1MDFmMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwM1QwMzI1MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMGZmOTdhNjY3NGI1NmU1YjM1MGZiNTA2ODcyMDBlYzY3ZWRiOWE0YTNiNGU1MjhlNmVhM2Q2YWY1ZGYxN2NjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yB1a5_SBoHuBI-oqKhF0C-k14VvbTU28YGlO1R5ZNuU)
> 
> The `EX`, `PX` etc... Am I right?

You only need to implement the GET option for this issus.

**Comment by yaten2302** [comment]
Ohh! Ok, now I got that.
earlier mykey was = "world";

So, when the user runs - `SET mykey "hello" GET`, it should return `world` (since, this was the previous key), am I right?

**Comment by yaten2302** [comment]
Hi @JyotinderSingh  , I apologize, actually, this issue, was inactive for 2 days 🙏 
I'm working on this issue, actually, I was trying to get familiar with the codebase. I've started working on this, and I'll create a draft PR by tomorrow 👍 

Just wanted to give an update that this issue is not inactive...

**Comment by yaten2302** [comment]
@JyotinderSingh , ig there are some issues with the [PR for this issue](https://github.com/DiceDB/dice/pull/246). I'll just close it and create a new one. 
Apologies for inconvenience 🙏

**Comment by arpitbbhayani** [comment]
Hello @yaten2302,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by yaten2302** [comment]
Hey @arpitbbhayani , yes, I'm working on this issue. Ig you also mentioned on the PR linked to this issue. Actually, due to some issues, I was inactive for a few days(I mentioned on Discord, too).

Also, for this issue, the NX option with the GET option is failing. I'll add some commits today for this. I'll surely ask if I've any doubt.

And, also, apologies for the delay🙏

**Comment by shardul08** [comment]
Hi @yaten2302,
Please feel free to share the issues you are facing with this implementation here, we would be happy to help.  Since this has been open for a long time now, let's try to close it soon.

Thanks for working on this!

**Comment by sanjaydev02** [comment]
@shardul08 its inactive for longtime can i take this issue?

**Comment by shardul08** [comment]
>  @shardul08 its inactive for longtime can i take this issue?

Let's wait for @yaten2302's response. You could later ask one of the maintainers to assign you the issue.

**Comment by sanjaydev02** [comment]
Hi @JyotinderSingh , I noticed that this issue has been inactive for some time. I'd be happy to pick it up and work on it. Please let me know if it's okay.

**Comment by JyotinderSingh** [comment]
> Hi @JyotinderSingh , I noticed that this issue has been inactive for some time. I'd be happy to pick it up and work on it. Please let me know if it's okay. 

Assigned to you @sanjaydev02

**Comment by JyotinderSingh** [comment]
#322 has almost all required changes, please refer to my last comments on that PR to see how to fix a few issues.
Please ensure you add a comprehensive test suite.

**Comment by sanjaydev02** [comment]
> > Hi @JyotinderSingh , I noticed that this issue has been inactive for some time. I'd be happy to pick it up and work on it. Please let me know if it's okay.
> 
> Assigned to you @sanjaydev02

Thanks @JyotinderSingh  for assigning this to me! I'll start working on it right away. I'll reach out if I encounter any issues. Looking forward to resolving this!

**Comment by sanjaydev02** [comment]
Hi @JyotinderSingh, I have some questions about my code. Can you help me when you have time? Thanks!"

**Comment by JyotinderSingh** [comment]
> Hi @JyotinderSingh, I have some questions about my code. Can you help me when you have time? Thanks!"

Please list the questions here.

**Comment by sanjaydev02** [comment]
when i'm adding case GET inside the switch case for checking if GET option is passed or not it throws error like its undefined.

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh when i'm adding case GET inside the switch case for checking if GET option is passed or not it throws error like its undefined. 

Please raise a draft PR, we can discuss there. It's hard to understand the issue with so little detail.

**Comment by sanjaydev02** [comment]
Hi @JyotinderSingh , I've created a draft PR ([link to PR](https://github.com/DiceDB/dice/pull/904)). Could you please check it and provide your feedback? Thank you!

**Comment by arpitbbhayani** [comment]
Hello @sanjaydev02,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by sanjaydev02** [comment]
@arpitbbhayani I’ve created a draft PR and would really appreciate your feedback so I can move forward. I’m still a beginner and any guidance from your side would be very helpful.

**Comment by sanjaydev02** [comment]
Hi @JyotinderSingh ,

I'm still working on this issue and wanted to give you a quick update. Apologies for the delay, I realize it's a simple one, and I'll have it resolved ASAP.

Thanks for your patience!

**Comment by sanjaydev02** [comment]
@JyotinderSingh hey i have some doubts can you look into ([PR](https://github.com/DiceDB/dice/pull/904))

**Comment by apoorvyadav1111** [comment]
Hi @sanjaydev02 , Hope you are well. I am eager to know more about any progress you have made on this issue.  As you know, we are looking forward to closing this issue, please reach to me or anyone else on the discord community if you need help.

**Comment by KanniShashankh** [comment]
@apoorvyadav1111 can i be assigned this issue? would love to work on it.

**Comment by sanjaydev02** [comment]
@apoorvyadav1111 hi i'm writing test for this one,feature is completed

**Comment by apoorvyadav1111** [comment]
Hi @KanniShashankh, Apologies, It was originally assigned to @sanjaydev02 and was unassigned as he closed his original PR. He recently communicated that he created a new PR with migrated logic.

**Comment by KanniShashankh** [comment]
No problem, i'll find another issue. 👍

**Comment by sanjaydev02** [comment]
https://github.com/DiceDB/dice/pull/1160

**Comment by tejasa97** [comment]
can i pick this up? since i see @sanjaydev02 having unassigned it from himself

**Comment by apoorvyadav1111** [comment]
Hi @tejasa97 , Thanks for sharing your interest on this issue. I am working on this one as this have been open for while. Will be raising a PR soon.
